x-pack/plugin/sql/qa/server/src/main/resources/math.sql-spec

```diff
@@ -125,7 +125,7 @@ roundInline3
 SELECT ROUND(-345.123, 0) AS rounded;
 
 roundInline4
-SELECT ROUND(-345.123,-51231231) AS rounded;
+SELECT ROUND(-345.123,-100) AS rounded;
```

Comment:
I think also H2 relies on something like `N / (10^-51231231)`, that returns NaN, but IMHO it's more a side effect of a poor implementation than a correct result. 
Now we return the right result (ie zero), so comparing it to H2 does not make much sense.

Reply:
It seems logical to me: since ROUND accepts any number as a second argument, it's supposed to handle it properly (and avoid overflows in the calculation algorithm). If `ROUND(n, x)` returns zero, also `ROUND(n, x-1)` should. IMHO it makes no much sense to have `ROUND(124, -323) == 0` and `ROUND(124, -324) == NaN`... (not a random example, it's what the previous implementation did, just because `Math.pow(10, -323) == 1.0E-323` and `Math.pow(10, -324) == 0.0`, that is an implementation detail, completely unrelated to the semantics of `ROUND` function)